### PR TITLE
Fused SDP runtime dispatch

### PR DESCRIPTION
**Pull request by drisspg**
# Summary 
This shows the run mechanism for dispatching to a custom backend kernel or defaulting to the composite version for SDP. Currently we short circuit for composite but in the future this will dispatch to the appropriate kernel.

cc @cpuhrsch @jbschlosser @bhosmer @mikaylagawarecki @erichan1

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -722,6 +723,62 @@ def set_weights_deterministic(model):
...
+        E = q.size(-1)
+        q = q / math.sqrt(E)
+        # (B, Nt, E) x (B, E, Ns) -> (B, Nt, Ns)
+        if attn_mask is not None:
+            attn = torch.baddbmm(attn_mask, q, k.transpose(-2, -1))
+        else:
+            attn = torch.bmm(q, k.transpose(-2, -1))
+
+        attn = torch.nn.functional.softmax(attn, dim=-1)
+        if dropout_p > 0.0:
```
Comment (vadimkantorov) [inline test/test_transformers.py:741]: btw I think inside dropout there's also a shortcut for p == 0.0, but not 100% sure if it incurs a copy or not

File: test/test_transformers.py [hunk 2]
```diff
@@ -722,6 +723,43 @@ def set_weights_deterministic(model):
         self.assertEqual(result.shape, ref_output.shape)
         torch.testing.assert_close(result, ref_output, atol=1e-3, rtol=1e-2)
 
+    @parametrize("type", ["dense", "nested"])
```
Comment (drisspg) [inline test/test_transformers.py:726]: @cpuhrsch bare minimum test added here but since we don't have any ci/cd setup to run this code. How do you think I should expand on these. Obviously wan't to add tests that check the dispatch mechanism
